move/interchain_token_service/sources/versioned/interchain_token_service_v0.move

```diff
@@ -181,6 +181,36 @@ module interchain_token_service::interchain_token_service_v0 {
         coin_management: CoinManagement<T>,
     ): TokenId {
         let token_id = token_id::from_coin_data(&self.chain_name_hash, &coin_info, &coin_management);
```

Comment:
This needs to be changed to pass in a bool. We can't rely on checking if coin metadata is stored in coin info anymore for the derivation